include/OnnxMlirRuntime.h

```diff
@@ -114,6 +121,16 @@
  * ```
  * Exactly as it should be.
  *
+ * \subsection freeing-tensor-memory Freeing Tensor Memory
+ *
+ *
+ * When creating an OMTensor or OMTensorList with a specified data pointer, a user has the option to
```

Comment:
We should seperate how OMTensor withOwnership works vs OMTensorList. This basically discribes how OMTensor withOwnership works but what is "owned" by on OMTensorList is much more intricate. (See earlier comment)

Reply:
Okay I separated them.....let me know if this is better. I try to summarize the words @AlexandreEichenberger stated in his examples below.